sklearn/feature_selection/rfe.py

```diff
@@ -376,23 +394,31 @@ def fit(self, X, y):
         scorer = check_scoring(self.estimator, scoring=self.scoring)
         n_features = X.shape[1]
         n_features_to_select = 1
+        rfe = RFE(estimator=self.estimator,
+                  n_features_to_select=n_features_to_select,
+                  step=self.step, verbose=self.verbose - 1)
 
-        # Determine the number of subsets of features
-        scores = []
 
+        # Determine the number of subsets of features
         # Cross-validation
-        for n, (train, test) in enumerate(cv.split(X, y)):
-            X_train, y_train = _safe_split(self.estimator, X, y, train)
-            X_test, y_test = _safe_split(self.estimator, X, y, test, train)
-
-            rfe = RFE(estimator=self.estimator,
-                      n_features_to_select=n_features_to_select,
-                      step=self.step, verbose=self.verbose - 1)
-
-            rfe._fit(X_train, y_train, lambda estimator, features:
-                     _score(estimator, X_test[:, features], y_test, scorer))
-            scores.append(np.array(rfe.scores_[::-1]).reshape(1, -1))
-        scores = np.sum(np.concatenate(scores, 0), 0)
+        # We do this separately for n_jobs=1 in order to be
+        # able to pass bound methods as scorers.
+        # If not, then joblib will raise a non-picklable error.
+        if self.n_jobs == 1:
```

Comment:
Providing something like this will fail with a non-picklable error if I don't branch (https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/feature_selection/tests/test_rfe.py#L166)